docs/reference/c++_support.rst

```diff
@@ -0,0 +1,173 @@
+.. meta::
+  :description: This chapter describes the C++ support of the HIP ecosystem
+  ROCm software.
+  :keywords: AMD, ROCm, HIP, C++
+
+*******************************************************************************
+C++ language support
+*******************************************************************************
+
+The ROCm platform enables the power of combined C++ and HIP (Heterogeneous-computing
+Interface for Portability) code. This code is compiled with a ``clang`` or ``clang++``
+compiler. The official compilers support the HIP platform, but for the most up-to-date
+feature set, use the ``amdclang`` or ``amdclang++`` components included in the ROCm
+installation.
+
+The source code is compiled according to the ``C++03``, ``C++11``, ``C++14``, ``C++17``,
+and ``C++20`` standards, along with HIP-specific extensions, but is subject to
+restrictions. The key restriction is the lack of standard library support. This is due to
+the HIP device's Single Instruction/Multiple Data (SIMD) nature, which makes most of the
```

Comment:
C++ standard libraries are supported for HIP host code. constexpr functions of C++ standard libraries are supported for HIP device code. Most functionality of C++ standard libraries are not supported for HIP device code since non-constexpr C++ functions are host functions, which cannot be called by device functions or kernels in HIP.